realm/realm-library/src/main/cpp/objectserver_shared.hpp

```diff
@@ -54,11 +54,13 @@ class JniSession {
             }
         };
         auto error_handler = [&, global_obj_ref_tmp](int error_code, std::string message) {
-            // FIXME: Simplify this by moving log_message to AndroidLogger
             JNIEnv *local_env;
             g_vm->AttachCurrentThread(&local_env, nullptr);
-            std::string log = num_to_string(error_code) + " " + message.c_str();
-            log_message(local_env, log_debug, log.c_str());
+            jclass java_session_class = local_env->GetObjectClass(global_obj_ref_tmp);
```

Comment:
@beeender If it's allowed to return null we have to check for it. At least assert().
Also goes for the next method.